Dockerfile

```diff
@@ -1,60 +1,47 @@
-FROM node:20.9 AS base
-FROM node:20.9-alpine AS base-alpine
-
-# build
-FROM base AS build
-
-RUN apt-get update && apt-get install git
-
-WORKDIR /root/app
-COPY package*.json ./
-# COPY prisma ./prisma/
-
-RUN npm ci
-
-COPY . ./
-
-RUN npx prisma generate --schema=./prisma/schema.prisma
-
-RUN npx prisma migrate
-
-RUN npm run build
-
-# development
-FROM base AS dev
-
-RUN apt-get update && apt-get install -y git \
-    postgresql-client \
-    curl \
-    vim
-
+FROM node:20.10-alpine AS base
+ENV NODE_ENV=production
+
+FROM base AS app-base
+RUN apk update \
+ && apk add tini\
+ && rm -rf /var/cache/apk/* \
+ && mkdir -p /home/node/app \
+ && chown -R node:node /home/node/app
+USER node
 WORKDIR /home/node/app
-COPY package*.json ./
+ENTRYPOINT [ "/sbin/tini", "--" ]
 
-RUN npm install
-COPY . ./
+FROM base AS build-base
+WORKDIR /build
+COPY package.json package-lock.json ./
+COPY prisma ./prisma/
+RUN npm ci --omi=dev --frozen-lockfile
 
+FROM build-base AS build
+COPY --chown=node:node . ./
 RUN npx prisma generate
+RUN npm run build
```

Comment:
Good question here actually. Should this really be two separate layers or just one? Meaning are these operations decoupled or they always need to be ran together?

More info on layers and what I mean: https://docs.docker.com/develop/develop-images/instructions/#run